R/Attributes.R

```diff
@@ -543,6 +543,23 @@ compileAttributes <- function(pkgdir = ".", verbose = getOption("verbose")) {
         list(env = list(PKG_CXXFLAGS ="-std=c++17"))
 }
 
+# built-in C++20 plugin for C++20
+.plugins[["cpp20"]] <- function() {
+    if (getRversion() >= "4.2")         # with recent R versions, R can decide
+        list(env = list(USE_CXX20 = "yes"))
+    else
+        list(env = list(PKG_CXXFLAGS ="-std=c++20"))
+}
+
+# built-in C++23 plugin for C++20
+.plugins[["cpp23"]] <- function() {
+    if (getRversion() >= "4.3")         # with recent R versions, R can decide
+        list(env = list(USE_CXX23 = "yes"))
+    else
+        list(env = list(PKG_CXXFLAGS ="-std=c++20"))
```

Comment:
I think it should be `PKG_CXXFLAGS ="-std=c++2b"` for experimental support.

Reply:
See previous comment.

Reply:
Here this is not the C++23 standard, and the `-std=c++23` flag doesn't exist yet, only the `c++2b` one. But I just checked, and it was added in gcc 11, so maybe it's too soon to add a C++23 plugin? Because there are still several machines with gcc 10 on CRAN. Maybe we could add a specific `cpp2b` plugin, as we did with `cpp2a`, and wait a bit for the official C++23 standard to come out for this one.

Reply:
On my machine, a few hours ago:

```
/usr/local/lib/R-devel/lib/R/bin/R CMD SHLIB --preclean -o 'sourceCpp_3.so' 'foo.cpp' 
ccache g++ -std=gnu++23 -I"/usr/local/lib/R-devel/lib/R/include" -DNDEBUG   -I"/usr/local/lib/R-devel/lib/R/library/Rcpp/include" -I"/tmp" -I/usr/local/include    -fpic  -ggdb -pipe -Wall -pedantic -flto -c foo.
cpp -o foo.o                                        
ccache g++ -std=gnu++23 -shared -ggdb -pipe -Wall -pedantic -flto -fpic -L/usr/local/lib/R-devel/lib/R/lib -L/usr/local/lib -o sourceCpp_3.so foo.o -L/usr/local/lib/R-devel/lib/R/lib -lR
> q()                            
edd@rob:~/git/rcpp(feature/c++23)$
```

I have not attempted to enable actual C++23 features -- that is for the compiler.  What I have attempted to enable in this PR is to use the C++23 compilation support _offered by R devel as of today_.  No more, no less.